pkg/resource/configoverrides.go

```diff
@@ -0,0 +1,13 @@
+package resource
+
+// ConfigOverrides holds data users can set to override default object configurations created
+// by this operator. This is stored in the registry Config.Spec.UnsupportedConfigOverrides.
+type ConfigOverrides struct {
+	Deploy *DeploymentOverrides `json:"deploy,omitempty"`
```

Comment:
It's better to name it as `deployment`, don't use abbreviations in API

Reply:
fixed.